public/components/common/custom-search-bar/custom-search-bar.tsx

```diff
@@ -49,51 +50,57 @@ export const CustomSearchBar = ({ filtersValues, ...props }) => {
     }
 
 
-    const buildCustomFilter = (isPinned: boolean, index?: any, querySearch?: any, key?: any): Filter => {
+    const buildCustomFilter = (isPinned: boolean, values?: any): Filter => {
+        const newFilters = []
+        values.forEach(element => {
+            const filter = {
+                match_phrase: {
+                    [element.value]: element.label
+                }
+            }
+            newFilters.push(filter);
+        });
+        const params = values.map(item => item.label)
         const meta: FilterMeta = {
             disabled: false,
             negate: false,
-            key: key,
-            params: { query: querySearch },
+            key: values[0].value,
+            params: params,
             alias: null,
-            type: "phrase",
-            index,
+            type: "phrases",
+            value: params.join(","),
+            index: AppState.getCurrentPattern(),
         };
         const $state: FilterState = {
             store: isPinned ? FilterStateStore.GLOBAL_STATE : FilterStateStore.APP_STATE,
         };
         const query = {
-            match_phrase: {
-                [key]: {
-                    query: querySearch
-                }
+            bool: {
+                minimum_should_match: 1,
+                should: newFilters
             }
         }
 
         return { meta, $state, query };
     };
 
     const setKibanaFilters = (values: any[]) => {
-        const newFilters = []
         const currentFilters = filterManager.getFilters().filter(item => item.meta.key != values[0].value)
         filterManager.removeAll()
         filterManager.addFilters(currentFilters)
-        values.forEach(element => {
-            const customFilter = buildCustomFilter(false, indexPattern.title, element.label, element.value)
-            newFilters.push(customFilter);
-        });
-        filterManager.addFilters(newFilters)
+        const customFilter = buildCustomFilter(false, values)
+        filterManager.addFilters(customFilter)
     }
 
     const refreshCustomSelectedFilter = () => {
         setSelectedOptions(defaultSelectedOptions)
-        const filters = filterManager.getFilters()
-        const filterCustom = filters.map(item => {
-            return {
-                value: item.meta.key,
-                label: item.meta.params.query,
-            }
-        }).filter(element => Object.keys(selectedOptions).includes(element.value))
+        const filterCustom = []
+        const filters = filterManager.getFilters().filter(item => item.meta.type === 'phrases' || Object.keys(selectedOptions).includes(item.meta.key)).map(element => ({ params: element.meta.params, key: element.meta.key })) || [];
+        filters.forEach(item => {
+            item.params.forEach(element => {
+                filterCustom.push({ label: element, value: item.key })
+            })
```

Comment:
**nitpick**: We could use `Array.prototype.map` instead of `Array.protype.forEach` + `Array.prototype.push`